src/core/tests/Test_AptitudePackageManager.py

```diff
@@ -349,6 +367,123 @@ def test_update_image_default_patch_mode_raises_exception(self):
+
+    def test_is_pro_client_prereq_met_should_return_false_for_unsupported_os_version(self):
+        package_manager = self.container.get('package_manager')
+        backup_envlayer_platform_linux_distribution = LegacyEnvLayerExtensions.LegacyPlatform.linux_distribution
+        backup_package_manager_ubuntu_pro_client_is_pro_working = package_manager.ubuntu_pro_client.is_pro_working
+        LegacyEnvLayerExtensions.LegacyPlatform.linux_distribution = self.mock_linux_distribution_to_return_ubuntu_focal
+        package_manager.ubuntu_pro_client.is_pro_working = self.mock_is_pro_working_return_true
+
+        self.assertFalse(package_manager.check_pro_client_prerequisites())
+
+        LegacyEnvLayerExtensions.LegacyPlatform.linux_distribution = backup_envlayer_platform_linux_distribution
+        package_manager.ubuntu_pro_client.is_pro_working = backup_package_manager_ubuntu_pro_client_is_pro_working
+
+    def test_is_pro_client_prereq_met_should_return_true_for_supported_os_version(self):
+        package_manager = self.container.get('package_manager')
+        backup_package_manager_ubuntu_pro_client_is_pro_working = package_manager.ubuntu_pro_client.is_pro_working
+        package_manager.ubuntu_pro_client.is_pro_working = self.mock_is_pro_working_return_true
+        backup_package_manager_is_minimum_required_python_installed = package_manager._AptitudePackageManager__is_minimum_required_python_installed
+        package_manager._AptitudePackageManager__is_minimum_required_python_installed = self.mock_minimum_required_python_installed_return_true
+
+        self.assertTrue(package_manager.check_pro_client_prerequisites())
+
+        package_manager.ubuntu_pro_client.is_pro_working = backup_package_manager_ubuntu_pro_client_is_pro_working
+        package_manager._AptitudePackageManager__is_minimum_required_python_installed = backup_package_manager_is_minimum_required_python_installed
+
+    def test_package_manager_instance_created_even_when_exception_thrown_in_pro(self):
+        package_manager = self.container.get('package_manager')
+        execution_config = self.container.get('execution_config')
+        backup_package_manager_ubuntu_pro_client_install_or_update_pro = UbuntuProClient.UbuntuProClient.install_or_update_pro
+        UbuntuProClient.UbuntuProClient.install_or_update_pro = self.mock_install_or_update_pro_raise_exception
+
+        obj = AptitudePackageManager.AptitudePackageManager(package_manager.env_layer, execution_config, package_manager.composite_logger, package_manager.telemetry_writer, package_manager.status_handler)
+
+        self.assertIsNotNone(obj)
+        self.assertIsNotNone(obj.ubuntu_pro_client)
+
+        UbuntuProClient.UbuntuProClient.install_or_update_pro = backup_package_manager_ubuntu_pro_client_install_or_update_pro
+
+    def mock_is_reboot_pending_returns_True(self):
+        return True, True
+
+    def mock_is_reboot_pending_returns_False(self):
+        return False, False
```

Comment:
nit: keep all mocked functions together, in this case before all tests

Reply:
Done.Thanks

Reply:
This seems to have been missed. 419 through 423 is supposed to be moved to around line 55.

Reply:
The checkin was not pushed. pushed it now.